src/main/java/zero/ui/Ui.java

```diff
@@ -0,0 +1,152 @@
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Displays the welcome message to the user.
+     */
+    public void showWelcome() {
+        System.out.println("____________________________________________________________");
+        System.out.println(" Hello! I'm zero.Zero");
+        System.out.println(" What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Displays the goodbye message to the user and sets the {@code isDone} flag to {@code true}.
+     */
+    public void showGoodbye() {
+        System.out.println("____________________________________________________________");
+        System.out.println(" Baibai!");
+        System.out.println("____________________________________________________________");
+        this.isDone = true;
+    }
+
+
+    public void showLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Displays an error message to the user.
+     *
+     * @param message The error message to display.
+     */
+    public void showError(String message) {
+        showLine();
+        System.out.println(" oopsie, " + message);
+        showLine();
+    }
+
+    /**
+     * Displays a message indicating that a task has been added, along with the updated task count.
+     *
+     * @param task The task that was added.
+     * @param taskCount The current number of tasks in the task list.
+     */
+    public void showTaskAdded(Task task, int taskCount) {
+        showLine();
+        System.out.println(" Got it. I've added this task:");
+        System.out.println("   " + task);
+        System.out.println(" Now you have " + taskCount + " tasks in the list.");
+        showLine();
+    }
+
+    /**
+     * Displays a message indicating that a task has been deleted, along with the updated task count.
+     *
+     * @param task The task that was deleted.
+     * @param taskCount The current number of tasks in the task list.
+     */
+    public void showTaskDeleted(Task task, int taskCount) {
+        showLine();
+        System.out.println(" Noted. I've removed this task:");
+        System.out.println("   " + task);
+        System.out.println(" Now you have " + taskCount + " tasks in the list.");
+        showLine();
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as done.
+     *
+     * @param task The task that was marked as done.
+     */
+    public void showTaskMarked(Task task) {
+        showLine();
+        System.out.println(" Nice! I've marked this task as done:");
+        System.out.println("   " + task);
+        showLine();
+    }
+
+    /**
+     * Displays a message indicating that a task has been unmarked (marked as not done).
+     *
+     * @param task The task that was unmarked.
+     */
+    public void showTaskUnmarked(Task task) {
```

Comment:
Naming of methods is nice and consistent, as well as, following the convention of being verbs! Well done!